Makefile

```diff
@@ -56,37 +50,35 @@ install: ## install from source (runs setup.py)
 	pip3 install -e .
 
 .PHONY: test
-test: clean local-tmp clone-dsp-repo dsp-stack ## run all tests
+test: dsp-stack ## run all tests
+	pytest test/
+	$(MAKE) stack-down
+
+.PHONY: test-no-stack
+test-no-stack: ## run tests without starting the stack (if a dsp-stack is already running)
 	pytest test/
 
 .PHONY: test-end-to-end
-test-end-to-end: clean local-tmp clone-dsp-repo dsp-stack ## run e2e tests
+test-end-to-end: dsp-stack ## run e2e tests
+	pytest test/e2e/
+	$(MAKE) stack-down
+
+.PHONY: test-end-to-end-no-stack
+test-end-to-end-no-stack: ## run e2e tests without starting the dsp-stack (if a dsp-stack is already running)
 	pytest test/e2e/
 
 .PHONY: test-unittests
 test-unittests: ## run unit tests
 	pytest test/unittests/
 
-.PHONY: local-tmp
```

Comment:
This is now included at the only place where it is necessary: before cloning DSP-API.